Sources/Commands/PackageCommands/EditCommands.swift

```diff
@@ -32,22 +32,33 @@ extension SwiftPackageCommand {
         @Option(help: "Create or use the checkout at this path")
         var path: AbsolutePath?
 
-        @Argument(help: "The name of the package to edit")
-        var packageName: String
+        @Argument(help: .hidden)
+        var packageName: String?
+
+        @Argument(help: "The identity of the package to edit.")
+        var packageIdentity: String = ""
```

Comment:
In theory yes - though since the order of arguments in this case would first assign `packageName` when parsing the incoming arguments, that would leave `packageIdentity` unassigned and would cause argument parser to throw an error. 

It seems like we could also just swap the placement of `packageName` and `packageIdentity` if we wanted to remove the default value, and to persist support for the use of `packageName` we could assign the value of `packageIdentity` to `packageName` instead, like so:

```
mutating func validate() throws { 
    if packageName == nil || packageName?.isEmpty { 
        packageName = packageIdentity
    }
}
```

Does this solution seem preferable?

Reply:
What's the reason for using `""` instead of `nil` as a default value in the first place?

Reply:
Particularly for `packageIdentity` in this case, I needed a property in the structure to be non-optional to avoid allowing the user to run `swift package edit` without any identifier for the package - although, I could just make it optional and use `nil` as a default value, and subsequently throw an error in the `run` method if that does happen :)

Reply:
> avoid allowing the user to run swift package edit without any identifier for the package
But isn't `""` an invalid identifier from that perspective, so for a user not passing `packageIdentity` explicitly, it is initialized with `""` and the end result is the same?

Reply:
Yeah, I wonder if we should just bite the bullet and do a forum post and propose changing this argument to be the identity instead of the name. Or can we autodetect if the name is an identity versus a name so it can do both? Use it as an identity but if that's not found use it as a name. If we can avoid making the user deal with this, that would be ideal.

Reply:
> But isn't "" an invalid identifier from that perspective, so for a user not passing packageIdentity explicitly, it is initialized with "" and the end result is the same?

When I was running this locally, since here `packageIdentity` isn't optional if I don't set a default value to it argument parser will throw an error. If a user passes in a package name (i.e. runs `swift package edit my-package-name`), it will assign the string first to `packageName` since this property is parsed first in the `Edit` command, but that leaves `packageIdentity` without a value (and here argument parser throws an error if that's the case) which is why I had initially set `packageIdentity` to `""` :)

Although, it seems like it may be better to have both `packageName` and `packageIdentity` set to `String?` with a default value of `nil`, and then execute the proper checks in the `validate` method to assure that there is an existing package name.

> I wonder if we should just bite the bullet and do a forum post and propose changing this argument to be the identity instead of the name.

That might be a good idea!

> Or can we autodetect if the name is an identity versus a name so it can do both?

There could be a way we can detect this - if we're able to find the package in the Workspace's dependencies, then it's the identifier. Otherwise, if we fetch its package manifest and compare it to the manifest `displayName`, that would tell us whether it's the package name.